File: 001_Two_Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = dict()
        for idx, num in enumerate(nums):
            if target - num in d:
                return [d[target - num], idx]
            d[num] = idx
        return []

File: 001_Two_Sum.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> dict;
        for (int i = 0; i < nums.size(); i++) {
            unordered_map<int, int>::iterator iter = dict.find(target - nums[i]);
            if (iter != dict.end()) {
                return vector<int>({iter->second, i});
            }
            dict[nums[i]] = i;
        }
        return vector<int>();
    }
};